android/cli/commands/_buildModule.js

```diff
@@ -59,12 +59,53 @@ AndroidModuleBuilder.prototype.validate = function validate(logger, config, cli)
 
 		this.manifest = this.cli.manifest;
 
-		const sdkModuleAPIVersion = this.cli.sdk && this.cli.sdk.manifest && this.cli.sdk.manifest.moduleAPIVersion && this.cli.sdk.manifest.moduleAPIVersion['android'];
-		if (this.manifest.apiversion && sdkModuleAPIVersion && this.manifest.apiversion !== sdkModuleAPIVersion) {
-			logger.error(__('The module manifest apiversion is currently set to %s', this.manifest.apiversion));
-			logger.error(__('Titanium SDK %s Android module apiversion is at %s', this.titaniumSdkVersion, sdkModuleAPIVersion));
-			logger.error(__('Please update module manifest apiversion to match Titanium SDK module apiversion.'));
-			process.exit(1);
+		const cliModuleAPIVersion = this.cli.sdk && this.cli.sdk.manifest && this.cli.sdk.manifest.moduleAPIVersion && this.cli.sdk.manifest.moduleAPIVersion['android'];
+		const needsMigration = this.manifest.apiversion && cliModuleAPIVersion && this.manifest.apiversion !== cliModuleAPIVersion;
+
+		// Migrate modules built for earlier versions of V8 / Titanium 
+		if (needsMigration) {
+			const cliSDKVersion = this.cli.sdk.manifest.version;
+			const manifestSDKVersion = this.manifest.minsdk;
+			const manifestModuleAPIVersion = this.manifest.apiversion;
+			const newVersion = String(parseInt(this.manifest.version.split('.')[0]) + 1) + '.0.0'; // TODO: Make this less error prone
+			const manifestTemplateFile = path.join(this.platformPath, 'templates', 'module', 'default', 'template', 'android', 'manifest.ejs');
+			
+			logger.info(__('⚠️  Detected Titanium %s that requires API-level %s, but the module currently only supports %s and API-level %s.', cliSDKVersion, cliModuleAPIVersion, manifestSDKVersion, manifestModuleAPIVersion)); 
+			logger.info(__('🛠  Migrating module manifest ...'));
+			
+			logger.info(__('🔄 Setting %s to %s', 'apiversion'.cyan, cliModuleAPIVersion.cyan));
+			this.manifest.apiversion = cliModuleAPIVersion;
+			
+			logger.info(__('🔄 Setting %s to %s', 'minsdk'.cyan, cliSDKVersion.cyan));
+			this.manifest.minsdk = cliSDKVersion;
+
+			logger.info(__('🔄 Bumping version from %s to %s', this.manifest.version.cyan, newVersion.cyan));
+			this.manifest.version = newVersion;
+			
+			// Pre-fill placeholders
+			var manifestContent = ejs.render(fs.readFileSync(manifestTemplateFile).toString(), {
+				moduleName: this.manifest.name,
+				moduleId: this.manifest.moduleid,
+				platform: this.manifest.platform,
+				tisdkVersion: this.manifest.minsdk,
+				guid: this.manifest.guid,
+				author: this.manifest.author,
+				publisher: this.manifest.author // The publisher does not have an own key in the manifest but can be different. Will override below
+			});
+						
+			// Migrate missing keys which don't have a placeholder (version, license, copyright & publisher) 
+			manifestContent = manifestContent.replace(/version.*/, 'version: ' + this.manifest.version);
```

Comment:
Maybe we could be more explicit in the matching? `/^version: \d+\.\d+\.\d+/m`, incase someone has the word version above the version key? Same for the two below?

Reply:
I tried to prevent  restricting the regex too much. I saw modules that have a version like "4.0" instead of "4.0.0" or other possible glitches. This attempts to be the "rule the all" way of doing it.

Reply:
That comment made me think, I could be wrong but doing semver.inc on '4.0' might return undefined/null as it wont be classed as valid semver. I think node-appc.version has a format function to pad a number to a certain number of segments that might be needed in that case